src/components/Map/Map.js

```diff
@@ -27,4 +22,10 @@ const Map = () => {
   )
 }
 
+async function validateAndFetchData(userPosition, setCoffeeShopsFn){      
+  if (apiService.validateCoordinates(userPosition.latitude, userPosition.longitude)) {
+    setCoffeeShopsFn(await apiService.getNearestShops(userPosition));
+  }
+}
```

Comment:
If this dataValidation function would only do validating, instead of also fetching data, you wouldn't need to pass the `setCoffeeShopsFn` to it.